cmd/ci-secret-generator/main_test.go

```diff
@@ -228,3 +229,47 @@ func TestValidateContexts(t *testing.T) {
 		})
 	}
 }
+func TestExecuteCommand(t *testing.T) {
+	testCases := []struct {
+		name          string
+		cmd           string
+		expected      []byte
+		expectedError error
+	}{
+		{
+			name:     "basic case",
+			cmd:      "echo basic case",
+			expected: []byte("basic case\n"),
+		},
+		{
+			name:          "error on no output",
+			cmd:           "true",
+			expectedError: fmt.Errorf("command \"true\" returned no output"),
+		},
+		{
+			name:          "error on cmd failure",
+			cmd:           "false",
+			expectedError: fmt.Errorf(" : exit status 1"),
+		},
+		{
+			name:          "error if stderr is not empty",
+			cmd:           ">&2 echo some error",
+			expectedError: fmt.Errorf("command \">&2 echo some error\" has error output"),
+		},
```

Comment:
Is this to be considered as an error even if the command return 0 as exit status?

Reply:
The reason is that it is likely that we need to handle the error message even if it returns 0.
The deprecation msg is an example.
We do not want to wait until it errors out to handle it. It would be too late and we would have pressure to look for solutions.
Instead, we want to handle it once we see the deprecation message, at the time, the command might still return 0.

Reply:
We didn't coordinate this, but I had the same idea.

This would allow us to catch future deprecation messages (or similar problems), which we would otherwise never see.  Unless we find some hostile program whose `stderr` we cannot control, I think this would be helpful.